.github/workflows/ci-pipeline.yml

```diff
@@ -26,30 +22,29 @@ jobs:
     runs-on: ubuntu-latest
     name: Get modified charts
     permissions:
-      contents: read
+      pull-requests: read
     outputs:
       chart: ${{ steps.get-chart.outputs.chart }}
       result: ${{ steps.get-chart.outputs.result }}
     steps:
-      - name: Install dependencies
-        run: sudo apt-get install -y patchutils
       - id: get-chart
         name: Get modified charts
         env:
-          DIFF_URL: "${{github.event.pull_request.diff_url}}"
-          TEMP_FILE: "${{runner.temp}}/pr-${{github.event.number}}.diff"
+          PULL_REQUEST_NUMBER: "${{ github.event.pull_request.number }}"
+          PULL_REQUEST_URL: "${{ github.event.pull_request.url }}"
+          GITHUB_TOKEN: "${{ github.token }}"
         run: |
-          # This request doesn't consume API calls.
-          curl -Lkso $TEMP_FILE $DIFF_URL
-          files_changed="$(sed -nr 's/[\-\+]{3} [ab]\/(.*)/\1/p' $TEMP_FILE | sort | uniq)"
+          # Using the Github API to detect the files changed as git merge-base stops working when the branch is behind
```

Comment:
Do we have data on current API usage and its potential increase with this change? Using `curl` was a conscious decision to avoid reaching the API rate limit. Be aware this step of the script is executed most of the time a PR is created/modified.

Reply:
I don't think there are other options besides the already used one. If the API call is still preferred, I would say we can go ahead with the change and monitor our CI's behaviour for a few days.